File: Commands/ShowMessage.cs
using GrapeCity.Forguncy.Commands;
using GrapeCity.Forguncy.Plugin;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ElementUI.Properties;

namespace ElementUI.Commands
{
    [Icon("pack://application:,,,/ElementUI;component/Resources/Images/Message.png")]
    [SRDisplayName(nameof(Resources.ShowMessage_ShowMessage))]
    public class ShowMessage : ELCommandBase
    {
        [FormulaProperty]
        [SearchableProperty]
        [SRDisplayName(nameof(Resources.ShowMessage_Message))]
        [DefaultValue(null)]
        [Required]
        public object Message { get; set; }

        [SearchableProperty]
        [SRDisplayName(nameof(Resources.ShowMessage_Type))]
        [SRComboProperty(ValueList = "success|warning|info|error", DisplayList = nameof(Resources.Type_DisplayList))]
        public string Type { get; set; } = "info";

        [FormulaProperty]
        [SearchableProperty]
        [SRDisplayName(nameof(Resources.ShowMessage_Duration))]
        [SRDescription(nameof(Resources.ShowMessage_Duration_Description))]
        public object Duration { get; set; }

        [FormulaProperty]
        [SearchableProperty]
        [SRDisplayName(nameof(Resources.ShowMessage_Offset))]
        [SRDescription(nameof(Resources.ShowMessage_Offset_Description))]
        [DefaultValue(20)]
        public object Offset { get; set; }

        [SRDisplayName(nameof(Resources.ShowMessage_ShowClose))]
        [DefaultValue(false)]
        public bool ShowClose { get; set; }

        [SRDisplayName(nameof(Resources.ShowMessage_Center))]
        [DefaultValue(false)]
        public bool Center { get; set; }

        public override string ToString()
        {
            return Resources.ShowMessage;
        }

        public override void InitDefaultPropertyValues()
        {
            base.InitDefaultPropertyValues();
            this.Message = Resources.ShowMessage_Message_DefaultValue;
            this.Duration = 3000;
            this.Offset = 20;
        }
    }
}

File: Resources/Commands/ShowMessage.ts
/// <reference path = "ELCommandBase.ts" />
namespace ElementCommands {
    interface ShowMessageParam {
        Center?: boolean;
        Duration?: string;
        Message?: string;
        Offset?: string;
        ShowClose?: boolean;
        Type?: string;
    }

    export class ShowMessage extends ElementCommandBase {

        public execute() {
            const { Center, Duration, Message, Offset, ShowClose, Type } = <ShowMessageParam>this.CommandParam;

            const duration = this.evaluateFormula(Duration) ?? 0;
            const offset = this.evaluateFormula(Offset) ?? 20;
            const type = this.evaluateFormula(Type);
            const message = this.evaluateFormula(Message) ?? "";
            const center = !!Center;
            const showClose = !!ShowClose;

            window.ElementPlus.ElMessage({
                duration,
                message,
                type,
                center,
                showClose,
                offset: Number(offset),
            });
        }
    }
}

Forguncy.Plugin.CommandFactory.registerCommand("ElementUI.Commands.ShowMessage, ElementUI", ElementCommands.ShowMessage);